clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -56,8 +56,14 @@ public int sizeInBytes() {
 
     @Override
     public long writeTo(GatheringByteChannel channel, long position, int length) throws IOException {
-        ByteBuffer dup = buffer.duplicate();
+        if (position > Integer.MAX_VALUE)
```

Comment:
Wondering if there's a good reason not to change the interface so that `position` is an integer? Similarly for the return type.